src/decisionengine/framework/modules/de_logger.py

```diff
@@ -55,79 +51,70 @@ def set_logging(
-            {
-                "class": "logging.handlers.RotatingFileHandler",
-                "maxBytes": max_file_size,
-                "backupCount": max_backup_count,
-            }
-        )
-        logconf.pylogconfig["handlers"]["file_structlog_debug"].update(
-            {
-                "class": "logging.handlers.RotatingFileHandler",
-                "maxBytes": max_file_size,
-                "backupCount": max_backup_count,
-            }
-        )
+        for files in logconf.de_outfile_info:
+            handler = logging.handlers.RotatingFileHandler(
+                filename=f"{log_file_name}" + files[0], maxBytes=max_file_size, backupCount=max_backup_count
+            )
+            handler.setLevel(files[1])
+            handler.setFormatter(logging.Formatter(files[2]))
+            handlers_list.append(handler)
 
     elif file_rotate_by == "time":
-        logconf.pylogconfig["handlers"]["de_file_debug"].update(
-            {
-                "class": "logging.handlers.TimedRotatingFileHandler",
-                "when": rotation_time_unit,
-                "interval": rotation_interval,
-            }
-        )
-        logconf.pylogconfig["handlers"]["de_file_info"].update(
-            {
-                "class": "logging.handlers.TimedRotatingFileHandler",
-                "when": rotation_time_unit,
-                "interval": rotation_interval,
-            }
-        )
-        logconf.pylogconfig["handlers"]["file_structlog_debug"].update(
-            {
-                "class": "logging.handlers.TimedRotatingFileHandler",
-                "when": rotation_time_unit,
-                "interval": rotation_interval,
-            }
-        )
+        for files in logconf.de_outfile_info:
+            handler = logging.handlers.TimedRotatingFileHandler(
+                filename=f"{log_file_name}" + files[0],
+                when=rotation_time_unit,
+                interval=rotation_interval,
+                backupCount=max_backup_count,
+            )
+            handler.setLevel(files[1])
+            handler.setFormatter(logging.Formatter(files[2]))
+            handlers_list.append(handler)
+
     else:
         raise ValueError(f"Incorrect 'file_rotate_by':'{file_rotate_by}:'")
 
-    logging.config.dictConfig(logconf.pylogconfig)
-    logger.debug("de logging setup complete")
+    structlog_handlers_list = [handlers_list.pop(i) for i in logconf.structlog_file_index]
+
+    # setup standard file handlers
+    for h in handlers_list:
+        delogger.addHandler(h)
+
+    # setup the queue logger
+    if unit_testing is False:
+        queue_logger.setup_queue_logging(delogger, structlog_handlers_list)
+        queue_logger.start()
+
+    delogger.debug("de logging setup complete")
 
 
 def get_logger():
     """
     get default logger - "decisionengine"
     :rtype: :class:`logging.Logger` - rotating file logger
     """
-    return logger
+    return delogger
```

Comment:
I believe this should be :

```suggestion
    nonlocal delogger
    return delogger
```